### DISCUSSION TITLE
Added experimental Depth control and preprocessing, using depth_anything model from LiheYoung.

### DISCUSSION THREAD

**Discussion by JackDainzh** [lllyasviel/Fooocus#2546] [Ideas]
So as the title says, I've decided to add an expanded control for personal use for now which allows for depth estimation and generation based on that. The feature was previously implemented in one the abandoned forks and used Midas for that.

I am listing the results below:

![image](https://github.com/lllyasviel/Fooocus/assets/44118724/72e678c7-880b-432f-bee8-06f0640ac9aa)
![image](https://github.com/lllyasviel/Fooocus/assets/44118724/478b4460-7010-48bf-a384-3a0b529053b5)

I conjunction with cpds or canny it can yield pretty good and fast results. The only downside it seems Fooocus doesn't have the cache store method for preprocessed images, so everytime generation starts it processes the image again, which in case with a large model such as depth_anything_vtl14 can be a bit annoying.

So the idea is, should I consider making a pull request and polishing it further, or should we just wait until the maintainer implements his version?

**Comment by brycegoh** [comment]
Hey, how did you add your own controlnet? Did you fork the repo?

**Reply by JackDainzh** [reply]
no, I just made a separate local repository. adding this is quite simple. come to think of it, any type of control net can be added respectively if you create a variable for it and manage how preprocessor works.
